realsense2_camera/src/rs_node_setup.cpp

```diff
@@ -215,7 +221,13 @@ void BaseRealSenseNode::startPublishers(const std::vector<stream_profile>& profi
         rmw_qos_profile_t qos = sensor.getQOS(sip);
         rmw_qos_profile_t info_qos = sensor.getInfoQOS(sip);
 
-        if (profile.is<rs2::video_stream_profile>())
+        // special handling for labeled point cloud stream
+        if (profile.is<rs2::video_stream_profile>() && profile.stream_type() == RS2_STREAM_LABELED_POINT_CLOUD)
+        {
+            _labeled_pointcloud_publisher = _node.create_publisher<sensor_msgs::msg::PointCloud2>("labeled_point_cloud/points",
+                rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(qos),qos));
+        }
+        else if (profile.is<rs2::video_stream_profile>())
```

Comment:
Since the video_stream_profile happens most of the time, it should be the first condition in the code.
Please pass it back to the if condition, and move the lpc condition to some else if after it (even maybe last one)

Reply:
Changed the flow.
Combined it to one  "if (profile.is<rs2::video_stream_profile>()) "
then asked about Labeled PC stream inside it.